operator/main.go

```diff
@@ -152,7 +161,11 @@ func runOperator(cmd *cobra.Command) {
 		enableCiliumEndpointSyncGC()
 	}
 
-	enableUnmanagedKubeDNSController()
+	if option.Config.DisableCiliumEndpointCRD {
+		log.Infof("KubeDNS unmanaged pods controller disabled as it requires %q option to be 'disabled' in Cilium ConfigMap", option.DisableCiliumEndpointCRDName)
```

Comment:
There is a lot of 'double-negative' logic here that is confusing me, because the 'disable-endpoint-crd' option has to be 'disabled', thus meaning that the CEP CRD has to be 'enabled' for the KubeDNS unmanaged pods controller to be enabled. Is that correct? I'd recommend saying: 
```
log.Infof("KubeDNS unmanaged pod controller disabled because %q option is set to \"true\"; to enable the kube-dns unmanaged pod controller, set %q to \"false\"", option.DisableCiliumEndpointCRDName, option.DisableCiliumEndpointCRDName)
```

Or alternatively:
```
log.Infof("CiliumEndpoint resource is disabled in Kubernetes, so it needs to be enabled via setting %s to \"false\" in the Cilium ConfigMap")
```

Reply:
I don't like the log message in general. Users will blindly enable it. Just state that it is disabled because CEP is disabled.